packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js

```diff
@@ -73,17 +73,38 @@ export default class Inspector extends PureComponent {
 
     state = {
         secondaryInspectorComponent: null,
-        toggledPanels: {}
+        toggledPanels: {},
+        viewConfiguration: null,
+        originalViewConfiguration: null
     };
 
+    configurationIsProcessed = false;
+
     constructor(props) {
         super(props);
-        this.cloneViewConfiguration(props);
+
+        if (props.focusedNode) {
+            const originalViewConfiguration = props.nodeTypesRegistry.getInspectorViewConfigurationFor($get('nodeType', props.focusedNode));
+            const nodeForContext = this.generateNodeForContext(
+                props.focusedNode,
+                props.transientValues
+            );
+
+            const processedViewConfiguration = this.preprocessViewConfiguration(
+                {node: nodeForContext}, [], {...originalViewConfiguration}, {...originalViewConfiguration}
```

Comment:
Please pardon my ignorance, but why is it necessary to clone (shallowly) the originalViewConfiguration here?
We are never mutating this object, we are using `produce`.
Besides, if we needed to clone it, why are we doing it shallowly? It won't save from deep state mutations.

And I'm sorry for missing your changes and taking so long to review them :(

Reply:
Hi @dimaip , don't worry about it! Just wanted to get it in as long as I remember a wee bit what I was doing here 🙈  
Maybe I don't  understand how `immer.js` works, but aren't we changing the reference here https://github.com/neos/neos-ui/pull/2571/files#diff-a7ae28cc7ab235343fc04fe7203f0b82R139 and here https://github.com/neos/neos-ui/pull/2571/files#diff-a7ae28cc7ab235343fc04fe7203f0b82R145

I removed the swallow copy in your marked line, because this not state (yet), but I feel safer leaving it in here: https://github.com/neos/neos-ui/pull/2571/files#diff-a7ae28cc7ab235343fc04fe7203f0b82R165-R166

I think a deep copy is not necessary since `produce` is it in place and we're only changing the reference.

Reply:
@gerhard-boden no, I don't think we are mutating anything. We are assigning a totally new object to the variable, that's not mutating the old object.
E.g. consider this code:
```
let a = {
first: 'object'
}
let b = a
a = {
  totally: 'different object'
}
console.log(b)
// {first: "object"}
```